#!/bin/sh /usr/share/dpatch/dpatch-run
## 01-argfix.dpatch
##
## DP: FIXME.

@DPATCH@

diff -Naurp dosfstools-2.11.orig/mkdosfs/mkdosfs.c dosfstools-2.11/mkdosfs/mkdosfs.c
--- dosfstools-2.11.orig/mkdosfs/mkdosfs.c	2005-03-12 16:12:16.000000000 +0000
+++ dosfstools-2.11/mkdosfs/mkdosfs.c	2008-06-26 12:35:45.000000000 +0000
@@ -1634,6 +1634,12 @@ main (int argc, char **argv)
   if (optind < argc)
     {
       device_name = argv[optind];  /* Determine the number of blocks in the FS */
+
+      if (!device_name) {
+	  printf("No device specified.\n");
+	  usage();
+      }
+
       if (!create)
          cblocks = count_blocks (device_name); /*  Have a look and see! */
     }
